api/v1/sp.go

```diff
@@ -68,6 +97,17 @@ func handleSuspendStorageProviders(c echo.Context) error {
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=GetStorageProvidersResponse}
 //	@Router			/sp/unsuspend [post]
-func handleUnsuspendStorageProvider(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleUnsuspendStorageProvider(c echo.Context) error {
+	var storageProviderIds Sps
+
+	err := json.NewDecoder(c.Request().Body).Decode(&storageProviderIds)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	var storageProviderResponse StorageProvidersResponse
+	a.db.Table("sps").Where("sp_id in (?)", storageProviderIds.SpId).UpdateColumn("tenant_sp_state", "active") //@jcace is active the right state
```

Comment:
Only valid to un-suspend an SP in the suspended state

it should to go ` db.TenantSpStateActive` after yes